tests/rptest/tests/topic_recovery_test.py

```diff
@@ -290,6 +290,77 @@ def verify_s3_content_after_produce(self):
         return False
 
 
+class EmptySegmentsCase(BaseCase):
+    """Restore topic that has segments in S3, but segments have only non-data 
+    batches (raft configuration, raft configuration).
+    """
+
+    topics = (TopicSpec(name='panda-topic',
+                        partition_count=1,
+                        replication_factor=3), )
+
+    def __init__(self, s3_client, kafka_tools, rpk_client, s3_bucket, logger,
+                 rpk_producer_maker, redpanda):
+        self._redpanda = redpanda
+        super(EmptySegmentsCase,
+              self).__init__(s3_client, kafka_tools, rpk_client, s3_bucket,
+                             logger, rpk_producer_maker)
+
+    def generate_baseline(self):
+        """Restart redpanda several times to get segments with no
+        data batches."""
+        for node in self._redpanda.nodes:
+            self._redpanda.stop_node(node)
+        time.sleep(1)
+        for node in self._redpanda.nodes:
+            self._redpanda.start_node(node)
+        time.sleep(1)
+        for node in self._redpanda.nodes:
+            self._redpanda.stop_node(node)
+        time.sleep(1)
+        for node in self._redpanda.nodes:
+            self._redpanda.start_node(node)
+        time.sleep(1)
+
+    def validate_node(self, host, baseline, restored):
+        """Validate restored node data using two sets of checksums.
+        The checksums are sampled from data directory before and after recovery."""
+        self.logger.info(f"Node: {baseline} checksums: {restored}")
+        # get rid of ntp part of the path
+        self.logger.info(
+            f"validate node host: {host}, baseline: {baseline}, restored: {restored}"
+        )
+        it = [os.path.basename(key) for key, _ in restored.items()]
+        assert len(it) == 1
+
+    def validate_cluster(self, baseline, restored):
+        """This method is invoked after the recovery and partition validation are
+        done.
+        Check that high_watermark is 0 for the new partition."""
+        def verify():
+            topic_state = list(self._rpk.describe_topic("panda-topic"))
```

Comment:
as an aside: I saw this describe_topic return NOT_LEADER_FOR_PARTITION.. in a test earlier where had to implement retry for this command, we might need a generic retry utility to loop until we get partition states in case a leader election is ongoing for use in all tests.